universe/marathon.json.mustache

```diff
@@ -46,49 +48,92 @@
     },
 {{/service.service_account_secret}}
 
-    "ports": [
-        0,
-        0,
-        0
+    "portDefinitions": [
+        {
+            "port": 0,
+            "protocol": "tcp",
+            "name": "dispatcher",
+            "labels": {
+                "VIP_0": "spark-dispatcher:7077"
+            }
+        },
+        {
+            "port": 0,
+            "protocol": "tcp",
+            "name": "dispatcher-ui",
+            "labels": {
+                "VIP_1": "spark-dispatcher:4040"
+            }
+        },
+        {
+            "port": 0,
+            "protocol": "tcp",
+            "name": "dispatcher-proxy",
+            "labels": {
+                "VIP_2": "spark-dispatcher:80"
+            }
+        }
     ],
+
+{{#service.UCR_containerizer}}
     "container": {
-        "type": "DOCKER",
-        "docker": {
-{{#service.docker-image}}
-            "image": "{{service.docker-image}}",
-{{/service.docker-image}}
-{{^service.docker-image}}
-            "image": "{{resource.assets.container.docker.spark_docker}}",
-{{/service.docker-image}}
-            "network": "HOST",
-{{#service.user}}
-            "parameters": [
-               {
-                 "key": "user",
-                 "value": "{{service.user}}"
-               }
-            ],
-{{/service.user}}
-            "forcePullImage": true
+    "type": "MESOS",
+    "docker": {
+    {{#service.docker-image}}
+        "image": "{{service.docker-image}}",
+    {{/service.docker-image}}
+    {{^service.docker-image}}
+        "image": "{{resource.assets.container.docker.spark_docker}}",
+    {{/service.docker-image}}
+    "forcePullImage": true
+    }
+    },
+{{/service.UCR_containerizer}}
+{{^service.UCR_containerizer}}
+    "container": {
+    "type": "DOCKER",
+    "docker": {
+    {{#service.docker-image}}
+        "image": "{{service.docker-image}}",
+    {{/service.docker-image}}
+    {{^service.docker-image}}
+        "image": "{{resource.assets.container.docker.spark_docker}}",
+    {{/service.docker-image}}
+    "network": "HOST",
+    {{#service.user}}
+        "parameters": [
+        {
+        "key": "user",
+        "value": "{{service.user}}"
         }
+        ],
+    {{/service.user}}
+    "forcePullImage": true
+    }
     },
+{{/service.UCR_containerizer}}
+
     "healthChecks": [
         {
-{{#security.ssl.enabled}}
-            "protocol": "COMMAND",
-            "command": { "value": "curl -f -k https://$HOST:$PORT2/" },
-{{/security.ssl.enabled}}
-{{^security.ssl.enabled}}
-            "path": "/",
             "portIndex": 2,
-            "protocol": "HTTP",
-{{/security.ssl.enabled}}
+            "protocol": "MESOS_HTTP",
```

Comment:
Just curious, is this change independent of the UCR change?

Reply:
I picked up this change from @vishnu2kmohan actually. It makes the health check in the Mesos UI show up green.

Reply:
Ok

Reply:
What is the difference?